src/project/project-actions.tsx

```diff
@@ -325,12 +335,54 @@ export class ProjectActions {
     }
   };
 
-  openIdea = async (idea: PythonProject) => {
-    if (await this.confirmReplace(idea.projectName)) {
-      await this.fs.replaceWithMultipleFiles(idea);
+  private openProject = async (
+    project: PythonProject,
+    confirmPrompt?: string
+  ) => {
+    if (await this.confirmReplace(confirmPrompt)) {
+      await this.fs.replaceWithMultipleFiles(project);
     }
   };
 
+  openIdea = async (slug: string | undefined, code: string, title: string) => {
+    this.logging.event({
+      type: "idea-open",
+      message: slug,
+    });
+    const pythonProject: PythonProject = {
+      files: projectFilesToBase64({
+        [MAIN_FILE]: code,
+      }),
+      projectName: title,
+    };
+    const confirmPrompt = this.intl.formatMessage(
+      { id: "confirm-replace-with-idea" },
+      { ideaName: pythonProject.projectName }
+    );
+    await this.openProject(pythonProject, confirmPrompt);
+    this.actionFeedback.success({
+      title: this.intl.formatMessage(
+        { id: "loaded-file-feedback" },
+        { filename: title }
+      ),
+    });
+  };
+
+  reset = async () => {
+    this.logging.event({
+      type: "reset-project",
+    });
+    await this.openProject(
+      defaultInitialProject,
+      this.intl.formatMessage({
+        id: "confirm-replace-reset",
+      })
+    );
+    this.actionFeedback.success({
+      title: this.intl.formatMessage({ id: "reset-project-feedback" }),
+    });
```

Comment:
This currently runs even if you clicked "Cancel"

Reply:
493724d287a5f984b1dacc52ea7aa92fd0e0d839